plugins/populator/frontend/public/javascripts/countly.models.js

```diff
@@ -394,7 +406,128 @@
+            }
+            event.segmentation.width = 1440;
+            event.segmentation.height = 3586;
+            event.segmentation.domain = window.location.origin;
+            event.segmentation.view = views[Math.floor(Math.random()*views.length)];
+            return [event];
+        }
+        this.getScrollmapEvents = function(){
+            var events = this.getHeatmapEvent();
+            if(Math.random() >= 0.5){
+                events = events.concat(this.getScrollmapEvent());
+                if (Math.random() >= 0.8) {
+                    events = events.concat(this.getScrollmapEvent());
+                }
+            }
+            return events;
+        }
+        this.getScrollmapEvent = function() {
+            this.stats.e++;
+            var views = ["/"+countlyCommon.ACTIVE_APP_KEY+"/demo-page.html"];
+            var event = {
+                "key": "[CLY]_action",
+                "count":1,
+                "timestamp":this.ts,
+                "hour":getRandomInt(0, 23),
+                "dow":getRandomInt(0, 6),
+                "test":1
+            }
+            this.ts += 1000;
+            event.segmentation = {};
+            event.segmentation.type = "scroll";
+            // 0: min value of scrollY variable for demoPage
+            // 3270: max value of scrollY variable for demoPage
+            // 983: viewportHeight
+            event.segmentation.y = getRandomInt(0, 3602) + 983;
+            event.segmentation.width = 1440;
+            event.segmentation.height = 3586;
+            event.domain = window.location.origin;
+            event.segmentation.view = views[Math.floor(Math.random()*views.length)];
+            return [event];
+        }
+        this.createFeedbackWidget = function() {
+            return $.ajax({
+                type: "GET",
+                url: countlyCommon.API_URL + "/i/feedback/widgets/create",
+                data: {
+                    api_key: countlyGlobal['member'].api_key,
+                    popup_header_text: chance.sentence({words:4}),
+                    popup_comment_callout: "Leave a comment",
+                    popup_email_callout: "Contact me by e-mail",
+                    popup_button_callout: "Send feedback",
+                    popup_thanks_message: "Thanks for feedback!",
+                    trigger_position: 'center-right',
+                    trigger_bg_color: '#13B94D',
+                    trigger_font_color: '#ffffff',
+                    trigger_button_text: "Feedback",
+                    target_devices: JSON.stringify(["desktop","phone","tablet"]),
+                    target_pages: JSON.stringify(["/"]),
+                    target_page: 'selected',
+                    is_active: true,
+                    hide_sticker: false,
+                    app_id: countlyCommon.ACTIVE_APP_ID
+                },
+                success: function (json, textStatus, xhr) {
+                }
```

Comment:
Don't we need widget id or something when providing feedback?
If no, then why we need to create a widget then?